packages/near-membrane-node/src/node-realm.ts

```diff
@@ -2,38 +2,40 @@ import {
     getFilteredEndowmentDescriptors,
     init,
     initSourceTextInStrictMode,
-    ProxyTarget,
+    DistortionCallback,
+    SupportFlags,
     VirtualEnvironment,
 } from '@locker/near-membrane-base';
 
 import { runInNewContext } from 'vm';
 
-interface EnvironmentOptions {
-    distortionCallback?: (originalTarget: ProxyTarget) => ProxyTarget;
+interface NodeEnvironmentOptions {
+    distortionCallback?: DistortionCallback;
     endowments?: object;
     globalThis: typeof globalThis;
+    support?: SupportFlags;
 }
 
 export default function createVirtualEnvironment(
-    options?: EnvironmentOptions
+    providedOptions?: NodeEnvironmentOptions
 ): (sourceText: string) => void {
-    const { distortionCallback, endowments, globalThis: blueGlobalThis = globalThis } = options || {
+    const options = {
         __proto__: null,
+        globalThis,
+        ...providedOptions,
     };
+    const { distortionCallback, endowments = {}, globalThis: blueGlobalThis, support } = options;
```

Comment:
Should `endowments = {}` be `endowments = { __proto__: null }`?

Reply:
Not needed because only own properties are gotten from it in `getFilteredEndowmentDescriptors` (uses `ReflectOwnKeys`).